### PR TITLE
React components

### PR DESCRIPTION
**Pull request by vadimshut**
1. Task: https://d14aa3aqnm30ti.cloudfront.net/docs/category/home-task
2. Screenshot:
image
3. Deploy:-
4. Done 23.06.2022 / deadline 27.06.2022
5. Score: 20 / 20

### CODE REVIEW

File: src/components/Courses/Courses.jsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+import './courses.scss';
+
+export const Courses = ({ coursesList, authorsList }) => {
+	const [courses, setCourses] = useState(coursesList);
+	const [filteredCourses, setFilteredCourses] = useState([]);
+	const [authors, setAuthors] = useState(authorsList);
+	const [shouldShowCreateCourse, setShouldShowCreateCourse] = useState(false);
+
+	useEffect(() => {
```
Comment (ArtemOsadchuck) [inline src/components/Courses/Courses.jsx]: Please use here useMemo

File: src/components/Courses/Courses.jsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+		setFilteredCourses(courses);
+	}, [courses]);
+
+	const handleClickSearch = (searchValue) => {
```
Comment (ArtemOsadchuck) [inline src/components/Courses/Courses.jsx]: Please wrap it in the useCallback

File: src/components/Courses/Courses.jsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+		const showCourses = courses.filter(
+			({ title, id }) =>
+				title.toLowerCase().includes(searchValue.toLowerCase()) ||
+				id.toLowerCase().includes(searchValue.toLowerCase())
+		);
+		setFilteredCourses(showCourses);
+	};
+
+	const updateAuthors = (oldAuthors, newAuthors) => {
```
Comment (ArtemOsadchuck) [inline src/components/Courses/Courses.jsx]: ditto

File: src/components/Courses/Courses.jsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+		const oldAuthorsIds = oldAuthors.map(({ id }) => id);
+		return newAuthors.filter(({ id }) => !oldAuthorsIds.includes(id));
+	};
+	const createNewCourse = (newCourse, newAuthors) => {
```
Comment (ArtemOsadchuck) [inline src/components/Courses/Courses.jsx]: ditto

**Review by ArtemOsadchuck — APPROVED** [review]

File: src/components/Courses/Courses.jsx (hunk 1 above)
Comment (vadimshut) [reply src/components/Courses/Courses.jsx]: add useMemo

File: src/components/Courses/Courses.jsx (hunk 2 above)
Comment (vadimshut) [reply src/components/Courses/Courses.jsx]: add useCallback

File: src/components/Courses/Courses.jsx (hunk 3 above)
Comment (vadimshut) [reply src/components/Courses/Courses.jsx]: add useCallback

File: src/components/Courses/Courses.jsx (hunk 4 above)
Comment (vadimshut) [reply src/components/Courses/Courses.jsx]: add useCallback

**Pull request merged**